### DISCUSSION TITLE
Latency issue for multi-region deployment

### DISCUSSION THREAD

**Discussion by anup-krai** [Kong/kong#7600] [Help] [answered]
We are running Kong in two regions with one common Postgres DB.
Region A - Kong with ingress controller, Postgres DB
Region B - Kong with ingress controller

Region B is also pointing to region A Postgres.

We have a simple API deployed in both the regions. When the call is triggered in Region A then Kong latency is around 4ms-8ms.
When the same call is triggered in Region B then Kong latency is around 30ms-40ms.

There are no security plugin added for this route. Why there is an increased latency in Region B. Is Kong calling DB for each transaction? As per the docs Kong caches the configuration in its memory from DB.

How can we remove the extra latency in region B?

**Comment by anup-krai** [comment]
Team - Any suggestions on this?

**Comment by kikito** [comment, marked as answer]
Hi, it's difficult to know exactly what's happening with the information you gave us. 

> Region B is also pointing to region A Postgres.

That sounds like the culprit, in general you should never have database servers in other regions. Depending on how you configured Region B, your Kong might be doing round trips between regions.

> Is Kong calling DB for each transaction?

Kong very aggressively caches database accesses. I am not sure I am understanding your setup correctly. It might be that the DB access is cached, but there is back-and-forth traffic between your ingress controller (in Region B) and your Kong instances (in Region A).

**Comment by hbagdi** [comment]
@Kong/team-k8s

**Comment by rainest** [comment]
We don't have much to add; functionally it's going to be the same as any multi-location database-backed deployment, so @kikito's answer is correct.

I agree that region B instances will be slower to warm cache. Being able to use a read replica in B instead may help (since they shouldn't need to write much of anything), but I'm not sure if we allow you to configure specific instances as the preferred instance for reads.
